runtime/interpreter/value.go

```diff
@@ -17556,6 +17577,7 @@ func (v *EphemeralReferenceValue) RemoveKey(
 	self := *referencedValue
 
 	interpreter.checkReferencedResourceNotDestroyed(self, getLocationRange)
+	v.checkReferenceResourceNotMoved(self, getLocationRange)
```

Comment:
These seem to occur together all the time; can we refactor this to be one function call that does both checks?

Reply:
Yeah, should be possible. I haven't migrated the existing tests, so wasn't sure what other places need updates, etc. So kept it separated for now. But I guess we can merge the two where ever possible.